server/mdm/microsoft/microsoft_mdm.go

```diff
@@ -0,0 +1,124 @@
+package microsoft_mdm
+
+import "github.com/fleetdm/fleet/v4/server/mdm/internal/commonmdm"
+
+const (
+	// MDMPath is Fleet's HTTP path for the core Microsoft MDM service.
+	MDMPath = "/api/mdm/microsoft"
+
+	// DiscoveryPath is the HTTP endpoint path that serves the IDiscoveryService functionality.
+	// This is the endpoint that process the Discover and DiscoverResponse messages
+	// See the section 3.1 on the MS-MDE2 specification for more details:
+	// https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-mde2/2681fd76-1997-4557-8963-cf656ab8d887
+	MDE2DiscoveryPath = MDMPath + "/discovery"
+
+	// MDE2PolicyPath is the HTTP endpoint path that delivers the X.509 Certificate Enrollment Policy (MS-XCEP) functionality.
+	// This is the endpoint that process the GetPolicies and GetPoliciesResponse messages
+	// See the section 3.3 on the MS-MDE2 specification for more details on this endpoint requirements:
+	// https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-mde2/8a5efdf8-64a9-44fd-ab63-071a26c9f2dc
+	// The MS-XCEP specification is available here:
+	// https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-xcep/08ec4475-32c2-457d-8c27-5a176660a210
+	MDE2PolicyPath = MDMPath + "/policy"
+
+	// MDE2EnrollPath is the HTTP endpoint path that delivers WS-Trust X.509v3 Token Enrollment (MS-WSTEP) functionality.
+	// This is the endpoint that process the RequestSecurityToken and RequestSecurityTokenResponseCollection messages
+	// See the section 3.4 on the MS-MDE2 specification for more details on this endpoint requirements:
+	// https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-mde2/5b02c625-ced2-4a01-a8e1-da0ae84f5bb7
+	// The MS-WSTEP specification is available here:
+	// https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-wstep/4766a85d-0d18-4fa1-a51f-e5cb98b752ea
+	MDE2EnrollPath = MDMPath + "/enroll"
+
+	// MDE2ManagementPath is the HTTP endpoint path that delivers WS-Trust X.509v3 Token Enrollment (MS-WSTEP) functionality.
+	// This is the endpoint that process the RequestSecurityToken and RequestSecurityTokenResponseCollection messages
+	// See the section 3.4 on the MS-MDE2 specification for more details:
+	// https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-mde2/5b02c625-ced2-4a01-a8e1-da0ae84f5bb7
+	MDE2ManagementPath = MDMPath + "/management"
+
+	// These are the entry points for the Microsoft Device Enrollment (MS-MDE) and Microsoft Device Enrollment v2 (MS-MDE2) protocols.
+	// These are required to be implemented by the MDM server to support user-driven enrollments
+	MSEnrollEntryPoint = "/EnrollmentServer/Discovery.svc"
+	MSManageEntryPoint = "/ManagementServer/MDM.svc"
```

Comment:
double checking mostly for my own understanding: are we required to name this two endpoints with these paths?

Reply:
This is a great question! The endpoints mentioned here are the default entry points specified in the MS-MDE2 and MS-MDM specs (enroll and management specs). These entry points are necessary for user-driven enrollments, where the user opens the Settings dashboard and manually enrolls themselves into an MDM by providing an email (UPN). In this scenario, the built-in OS MDM Enrollment Client will contact the Fleet MDM server using these endpoints as the entry points for the protocols. Currently, we are working on supporting programmatic enrollment (where we can control the entry points), but I believe creating endpoint aliases using these values is a minor change that we can make to start supporting user-driven enrollments. So I left these here and I'm planning to tackle the aliases work [here](https://github.com/fleetdm/fleet/issues/12264)